components/cluster/corosync/Makefile

```diff
@@ -0,0 +1,80 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL)". You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2013 Adam Stevko. All rights reserved.
+# Copyright 2016 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME= corosync
+COMPONENT_VERSION= 2.4.1
+COMPONENT_FMRI= application/cluster/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION= System/Services
+COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE= v$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH= sha256:0fae5f07af8854c8921f243469be9052e3795237f5c6293d12ae235d2f75e965
+COMPONENT_PROJECT_URL= https://github.com/corosync/corosync
+COMPONENT_ARCHIVE_URL= https://github.com/$(COMPONENT_NAME)/$(COMPONENT_NAME)/archive/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE= BSD
+COMPONENT_SUMMARY= The Corosync cluster engine.
+
+PATH= /usr/gnu/bin:/usr/bin:/usr/sbin:/sbin
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+PATCH_LEVEL = 1
+
+COMPONENT_PREP_ACTION=(cd $(SOURCE_DIR); $(CONFIG_SHELL) autogen.sh )
+
+COMPONENT_PRE_CONFIGURE_ACTION=($(CLONEY) $(SOURCE_DIR) $(@D))
+
+CONFIGURE_SCRIPT= $(@D)/configure
+
+CFLAGS += -D__EXTENSIONS__ -D_POSIX_PTHREAD_SEMANTICS
+LDFLAGS += -R/usr/lib/mps
+
+CONFIGURE_OPTIONS += --sysconfdir=/etc
+CONFIGURE_OPTIONS += --libexecdir=$(CONFIGURE_PREFIX)/lib
+CONFIGURE_OPTIONS += --sharedstatedir=/var/cluster
```

Comment:
/var/cluster vs /var/run/hacluster (different values), we should possibly unite this.